tools/make/golang.mk

```diff
@@ -56,17 +56,20 @@ go.testdata.complete: ## Override test ouputdata
 	go test -timeout 60s github.com/envoyproxy/gateway/internal/gatewayapi --override-testdata=true
 
 .PHONY: go.test.coverage
-go.test.coverage: $(tools/setup-envtest) ## Run go unit and integration tests in GitHub Actions
+go.test.coverage: go.test.cel ## Run go unit and integration tests in GitHub Actions
 	@$(LOG_TARGET)
 	KUBEBUILDER_ASSETS="$(shell $(tools/setup-envtest) use $(ENVTEST_K8S_VERSION) -p path)" \
-		go test ./... --tags=integration,celvalidation -race -coverprofile=coverage.xml -covermode=atomic
+		go test ./... --tags=integration -race -coverprofile=coverage.xml -covermode=atomic
 
 .PHONY: go.test.cel
 go.test.cel: manifests $(tools/setup-envtest) # Run the CEL validation tests
 	@$(LOG_TARGET)
-	go clean -testcache # Ensure we're not using cached test results
-	KUBEBUILDER_ASSETS="$(shell $(tools/setup-envtest) use $(ENVTEST_K8S_VERSION) -p path)" \
-		go test ./test/cel-validation --tags=celvalidation -race
+	@for ver in $(ENVTEST_K8S_VERSIONS); do \
```

Comment:
`!has(self.backendRef) || !has(self.backendRef.kind) || self.backendRef.kind == 'Service'` passed on `1.27.1` and failed on `1.28.0`

so we need run CEL on different version of kubernetes.